pshtt/pshtt.py

```diff
@@ -99,7 +100,41 @@ def result_for(domain):
         'HSTS Max Age': hsts_max_age(domain),
         'HSTS Entire Domain': is_hsts_entire_domain(domain),
         'HSTS Preload Ready': is_hsts_preload_ready(domain),
-        'HSTS Preloaded': is_hsts_preloaded(domain)
+        'HSTS Preloaded': is_hsts_preloaded(domain),
+
+        # A domain 'Supports HTTPS' when it doesn't downgrade and has valid HTTPS,
+        # or when it doesn't downgrade and has a bad chain but not a bad hostname.
+        # Domains with a bad chain "support" HTTPS but user-side errors should be expected.
+        'Domain Supports HTTPS': (
+            (is_downgrades_https(domain) != True) and
+            is_valid_https(domain)
+        ) or (
+            (is_downgrades_https(domain) != True) and
+            is_bad_chain(domain) and
+            (is_bad_hostname(domain) != True)
+        ),
```

Comment:
I recommend (a) moving this boolean code to its own method (like the other methods are), and (b) reusing method calls to `is_defaults_to_https` and etc as needed, rather than copying the same logic here. This will make the code more maintainable and consistent.